compaction/compaction_manager.cc

```diff
@@ -756,15 +758,24 @@ future<> compaction_manager::postponed_compactions_reevaluation() {
             _postponed.clear();
             co_return;
         }
-        auto postponed = std::move(_postponed);
+        // A task_state being reevaluated can re-insert itself into postponed list, which is the reason
```

Comment:
Note: correct use of condition_variable::when() should have a predicate that checks for _state and non-empty queue.